plugins/action_generate_review_assignment_records/src/generateReviewAssignments.ts

```diff
@@ -138,12 +138,13 @@ const generateNextReviewAssignments = async ({
   // Build reviewers into object map so we can combine duplicate user_orgs
   // and merge their section code restrictions
   nextLevelReviewers.forEach((reviewer: Reviewer) => {
-    const { userId, orgId, allowedSections, canSelfAssign } = reviewer
+    const { userId, orgId, allowedSections, canSelfAssign, canMakeFinalDecision } = reviewer
 
-    const status =
-      canSelfAssign || nextReviewLevel > 1
-        ? ReviewAssignmentStatus.AvailableForSelfAssignment
-        : ReviewAssignmentStatus.Available
+    const status = canMakeFinalDecision
```

Comment:
I find nested inline condition statements really hard to read. not a required change, since it's a stylistic preference, but would prefer something like:
```JS
getAssignmentStatus = () => {
  if(canMakeFinalDecision) return ReviewAssignmentStatus.Assigned;
  if(canSelfAssign || nextReviewLevel > 1) return ReviewAssignmentStatus.AvailableForSelfAssignment
  return ReviewAssignmentStatus.Available
}
```
above would also help make it extendable if more conditions needs to be added

Reply:
Yes, good point. Done: e135430